src/VisualStudio/DevKit/Impl/EditAndContinue/DynamicDocumentSyncRegistration.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Roslyn.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices.DevKit.EditAndContinue;
+
+[ExportCSharpVisualBasicLspServiceFactory(typeof(OnInitialized)), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class DynamicDocumentSyncRegistration() : ILspServiceFactory
```

Comment:
I think I'd prefer this to be in the main language server project.  We can check if devkit is enabled using a global option, but maybe we should just have it always be on too.

Reply:
Do you just want it next to the sync handlers?

Reply:
If we put it in the protocol project, it will light up in VS.  If you want that, then yes.  Otherwise put it in the LanguageServer.csproj project (probably in the language server folder).

This is the devkit global option - https://github.com/dotnet/roslyn/blob/main/src/Features/LanguageServer/Protocol/LspOptionsStorage.cs#L31